### PR TITLE
proto/orijtech: define per point data

### PR DESCRIPTION
**Pull request by odeke-em**
Updates the protobuf definitions to have per point data that can then be used by the user interface.

Updates #14
Updates #16

### CODE REVIEW

**Review by kirbyquerby — CHANGES_REQUESTED** [review]

**Comment by willpoint** [comment]
@odeke-em @kirbyquerby how do we proceed on this, so I can update the frontend as well.

File: proto/orijtech/cosmosloadtester/v1/loadtest_service.proto [hunk 1]
```diff
@@ -96,4 +96,29 @@ message RunLoadtestResponse {
...
+  double bytes = 3;
+  Ranking bytes_rankings = 4;
+  Ranking latency_rankings = 5;
+}
+
+message Percentile {
+  google.protobuf.Duration at = 1;
+  google.protobuf.Duration latency = 2;
+  int64 size = 3;
+  string at_str = 4;
```
[the remark this answers is not in the document]
Comment (odeke-em) [reply proto/orijtech/cosmosloadtester/v1/loadtest_service.proto:128]: Human friendly representation of when the value occured. Pretty useful for debugging. I've documented it.

File: proto/orijtech/cosmosloadtester/v1/loadtest_service.proto [hunk 2]
```diff
@@ -96,4 +96,29 @@ message RunLoadtestResponse {
   // The rate at which data was transmitted in transactions (bytes/sec).
   // Corresponds to avg_data_rate in tm-load-test.
   double avg_bytes_per_second = 5;
+
+  repeated PerSecond per_sec = 6;
```
[the remark this answers is not in the document]
Comment (odeke-em) [reply proto/orijtech/cosmosloadtester/v1/loadtest_service.proto:101]: Done, thanks @kirbyquerby!

File: proto/orijtech/cosmosloadtester/v1/loadtest_service.proto [hunk 3]
```diff
@@ -96,4 +96,29 @@ message RunLoadtestResponse {
...
+  int64 sec = 1;
+  double qps = 2;
+  double bytes = 3;
+  Ranking bytes_rankings = 4;
+  Ranking latency_rankings = 5;
+}
+
+message Percentile {
+  google.protobuf.Duration at = 1;
+  google.protobuf.Duration latency = 2;
```
[the remark this answers is not in the document]
Comment (odeke-em) [reply proto/orijtech/cosmosloadtester/v1/loadtest_service.proto:124]: Percentile can have both latency or size, I suppose we can use a oneof type for these but it is much simpler to have them both; when we have bytes_rankings, latency is unset, when we have latency_rankings, size is unset

File: proto/orijtech/cosmosloadtester/v1/loadtest_service.proto [hunk 4]
```diff
@@ -96,4 +96,29 @@ message RunLoadtestResponse {
   // The rate at which data was transmitted in transactions (bytes/sec).
   // Corresponds to avg_data_rate in tm-load-test.
   double avg_bytes_per_second = 5;
+
+  repeated PerSecond per_sec = 6;
+}
+
+message PerSecond {
+  int64 sec = 1;
```
[the remark this answers is not in the document]
Comment (odeke-em) [reply proto/orijtech/cosmosloadtester/v1/loadtest_service.proto:109]: It is!

[the remark this answers is not in the document]
Comment (odeke-em) [reply proto/orijtech/cosmosloadtester/v1/loadtest_service.proto:109]: No, it is an ordinal number for example sec=0, sec=10 and essentially the lower bound of time bucket.

File: proto/orijtech/cosmosloadtester/v1/loadtest_service.proto [hunk 5]
```diff
@@ -96,4 +96,29 @@ message RunLoadtestResponse {
...
+  double qps = 2;
+  double bytes = 3;
```
[the remark this answers is not in the document]
Comment (odeke-em) [reply proto/orijtech/cosmosloadtester/v1/loadtest_service.proto]: Done, thanks @kirbyquerby!

File: proto/orijtech/cosmosloadtester/v1/loadtest_service.proto [hunk 6]
```diff
@@ -96,4 +96,29 @@ message RunLoadtestResponse {
...
+  Ranking bytes_rankings = 4;
+  Ranking latency_rankings = 5;
+}
+
+message Percentile {
+  google.protobuf.Duration at = 1;
```
[the remark this answers is not in the document]
Comment (odeke-em) [reply proto/orijtech/cosmosloadtester/v1/loadtest_service.proto]: I've updated it to start_offset

**Comment by odeke-em** [comment]
Thanks for the review @kirbyquerby! Updated with the AIP suggestions. @willpoint thanks for the ping! We are almost there, I've made some changes and regenerated the protobuf. Just another round of reviews!

File: proto/orijtech/cosmosloadtester/v1/loadtest_service.proto (hunk 4 above)
[the remark this answers is not in the document]
Comment (kirbyquerby) [reply proto/orijtech/cosmosloadtester/v1/loadtest_service.proto:109]: So this is the duration (floored to the nearest second) relative to the start of the load test.

**Review by kirbyquerby — APPROVED** [review]
Approving because of timeline constraints.

File: proto/orijtech/cosmosloadtester/v1/loadtest_service.proto [hunk 7]
```diff
@@ -96,4 +96,47 @@ message RunLoadtestResponse {
   // The rate at which data was transmitted in transactions (bytes/sec).
   // Corresponds to avg_data_rate in tm-load-test.
   double avg_bytes_per_second = 5;
+
+  // The respective points per second from 0 until the request's max_time.
+  repeated PerSecond per_sec = 6;
```
Comment (kirbyquerby) [inline proto/orijtech/cosmosloadtester/v1/loadtest_service.proto:101]: @odeke-em in chat, you mentioned:
> about the slice: we might have scenarios in which there is more than 1 connection so safe to return it as a slice. For starters let’s let it display stats per transactor. Once the app is being used we shall have breathing space to display an aggregation

This proto definition only returns aggregate values, while the custom tm-load-test returns a `[]*ProcessedStats`, with an entry for each transactor:

https://github.com/orijtech/tm-load-test/blob/d37154798c88c311e880eb23ec799d09e04cb44b/pkg/loadtest/loadtest.go#L15

Should the proto be changed or should the implementation aggregate the stats?

**Review by willpoint — APPROVED** [review]
LGTM

**Comment by kirbyquerby** [comment]
Merging to unblock UI.

**Pull request merged**